rasa/core/training/story_conflict.py

```diff
@@ -304,17 +304,28 @@ def _get_previous_event(
     # A typical state is, for example,
     # `{'prev_action_listen': 1.0, 'intent_greet': 1.0, 'slot_cuisine_0': 1.0}`.
     # We need to look out for `prev_` and `intent_` prefixes in the labels.
-    for turn_label in state:
-        if (
-            turn_label.startswith(PREV_PREFIX)
-            and turn_label.replace(PREV_PREFIX, "") != ACTION_LISTEN_NAME
-        ):
-            # The `prev_...` was an action that was NOT `action_listen`
-            return "action", turn_label.replace(PREV_PREFIX, "")
-        elif turn_label.startswith(INTENT + "_"):
+    for origin, substate in state.items():
+        if origin == PREVIOUS_ACTION:
+            if (
+                "action_name" in substate
+                and substate["action_name"] != ACTION_LISTEN_NAME
+            ):
+                # The `prev_...` was an action that was NOT `action_listen`
+                assert isinstance(substate["action_name"], str)
```

Comment:
I think it would be better to do a proper check and raise an exception with a good explanation if the action name is not a string. In what kind of situations might action name be not a string?

Reply:
I think action name cannot be anything but a string. It's just that `Substate` doesn't require that in terms of types.

Reply:
So I'd think that if this assertion fails, somebody must have modified the Rasa source code.

Reply:
Is there a way to ensure the type of the value of one key (`action_name`) is actually `Text`?

Reply:
No, as far as I know you cannot make the type dependent on a specific value. 
I would nevertheless go with raising a specific exception instead of the assertion. That makes it also more clear to the users and the developers. You could maybe also add a comment why this exception is raised.

Reply:
ok, will do :)